### PR TITLE
Support split operation

### PR DESCRIPTION
**Pull request by fujunwei**

### CODE REVIEW

File: src/webnn_native/dml/GraphDML.cpp [hunk 1]
```diff
@@ -1092,6 +1092,31 @@ namespace webnn_native { namespace dml {
...
+        DAWN_ASSERT(mExpression.find(inputOperand) != mExpression.end());
+        ::dml::Expression input = mExpression.at(inputOperand);
+        ::dml::TensorDimensions inputDims = input.GetOutputDesc().sizes;
+
+        // dml::Span just holds the refernces, need a variable to hold the memory.
+        std::vector<uint32_t> splits = split->GetSplits();
+        int32_t axis = split->GetAxis();
+        if (splits.size() == 1) {
+            if (inputDims[axis] % splits[0] != 0) {
+                return DAWN_INTERNAL_ERROR("The splits can't be splitted.");
```
Comment (huningxin) [inline src/webnn_native/dml/GraphDML.cpp:1107]: I would suggest the error message could be specific, such as "the axis x with size n can't be divisible by splits[0] m" where x, n, m are actual values.

File: webnn.json [hunk 2]
```diff
@@ -143,7 +143,20 @@
     "category": "object"
   },
   "operator": {
-    "category": "object"
+    "category": "object",
+    "methods": [
```
Comment (huningxin) [inline webnn.json:147]: Why do we need to expose these two methods as public APIs?

**Comment by fujunwei** [comment]
Done thanks, PTAL.

File: webnn.json [hunk 3]
```diff
@@ -157,6 +157,22 @@
       }
     ]
   },
+  "array operand": {
```
Comment (huningxin) [inline webnn.json:160]: "operand array"?

File: webnn.json [hunk 4]
```diff
@@ -157,6 +157,22 @@
...
+    "category": "object",
+    "methods": [
+      {
+        "name": "size",
+        "returns": "int32_t"
```
Comment (huningxin) [inline webnn.json:165]: size_t?

File: webnn.json [hunk 5]
```diff
@@ -157,6 +157,22 @@
...
+      },
+      {
+        "name": "get",
+        "returns": "operand",
+        "args": [
+          {"name": "index", "type": "int32_t"}
```
Comment (huningxin) [inline webnn.json:171]: size_t?

File: webnn.json (hunk 3 above)
Comment (fujunwei) [reply webnn.json]: Done, thanks.

**Review by huningxin — APPROVED** [review]
LGTM, thanks much @fujunwei !

File: src/webnn_native/openvino/GraphIE.cpp [hunk 6]
```diff
@@ -801,6 +807,36 @@ namespace webnn_native { namespace ie {
...
+            ngraph_node_t* splitsNode =
+                AddConstantWithGraph<uint32_t>(precision_e::U32, {splits.size()}, splits);
+            status = ngraph_variadic_split(input, axisNode, splitsNode, &outputNodes);
+            ngraph_node_free(&splitsNode);
+        }
+        DAWN_TRY(CheckStatusCode(status, "ngraph split"));
+        uint32_t number;
+        status = ngraph_get_output_number(outputNodes, &number);
+        DAWN_TRY(CheckStatusCode(status, "ngraph get output number"));
+        for (uint32_t i = 0; i < number; ++i) {
```
Comment (mingmingtasd) [inline src/webnn_native/openvino/GraphIE.cpp:830]: Need we check `number == split->output().szie()`?

File: third_party/openvino/ngraph_c_api/src/ngraph_c_api.cpp [hunk 7]
```diff
@@ -50,7 +52,7 @@
   CREATE_NODE_AND_CATCH_EXCEPTIONS(impl, node)
 
 struct ngraph_node {
```
Comment (miaobin) [inline third_party/openvino/ngraph_c_api/src/ngraph_c_api.cpp:54]: Can we name the structure 'ngraph_output' or 'ngraph_output_node'?

File: src/webnn_native/openvino/GraphIE.cpp (hunk 6 above)
Comment (fujunwei) [reply src/webnn_native/openvino/GraphIE.cpp:830]: That's good assert, done, thanks.

File: third_party/openvino/ngraph_c_api/src/ngraph_c_api.cpp (hunk 7 above)
Comment (fujunwei) [reply third_party/openvino/ngraph_c_api/src/ngraph_c_api.cpp:54]: That's a good discussion. I don't want to export `ngraph_output_node_t `c struct that will confuse to read code, so we consider the `ngraph_node_t `is output node, but it's still an open issue.
```
IEStatusCode ngraph_variadic_split(const ngraph_node_t* input,
                                   const ngraph_node_t* axis,
                                   const ngraph_node_t* splits,
                                   ngraph_node_t** node) {
```

=>

```
IEStatusCode ngraph_variadic_split(const ngraph_output_node_t * input,
                                   const ngraph_output_node_t * axis,
                                   const ngraph_output_node_t * splits,
                                   ngraph_output_node_t ** node) {
```

File: src/webnn_native/dml/GraphDML.cpp [hunk 8]
```diff
@@ -1102,6 +1102,34 @@ namespace webnn_native { namespace dml {
...
+                return DAWN_INTERNAL_ERROR("the axis " + std::to_string(axis) + " with size " +
+                                           std::to_string(splits.size()) +
+                                           " can't be divisible by splits[0] " +
+                                           std::to_string(splits[0]) + ".");
+            }
+            splits = std::vector(splits[0], inputDims[axis] / splits[0]);
+        }
+        ::dml::Span<const uint32_t> splitsSpan(splits);
+        std::vector<::dml::Expression> output = ::dml::Split(input, axis, splitsSpan);
+        size_t outputSize = split->Outputs().size();
```
Comment (mingmingtasd) [inline src/webnn_native/dml/GraphDML.cpp:1126]: Also need check output.size() == outputSize? Thanks!

Comment (fujunwei) [reply src/webnn_native/dml/GraphDML.cpp:1126]: I think it's a good code style, i prefer adding the assert, done, thanks.

**Review by mingmingtasd — APPROVED** [review]
LGTM,thanks! @fujunwei

File: src/webnn_native/dml/GraphDML.cpp (hunk 8 above)
Comment (miaobin) [reply src/webnn_native/dml/GraphDML.cpp:1126]: Got it and we may need to adding the different asserts based on different actual cases.
LGTM, thanks! @fujunwei

File: third_party/openvino/ngraph_c_api/src/ngraph_c_api.cpp (hunk 7 above)
Comment (miaobin) [reply third_party/openvino/ngraph_c_api/src/ngraph_c_api.cpp:54]: Indeed, it is easy to confuse code reading. Maybe we can find a suitable way to sort out the naming in the following discussing.

**Review by miaobin — APPROVED** [review]

File: third_party/openvino/ngraph_c_api/src/ngraph_c_api.cpp (hunk 7 above)
Comment (fujunwei) [reply third_party/openvino/ngraph_c_api/src/ngraph_c_api.cpp:54]: Great, thanks. merge it first.

**Pull request merged**